addons/calendar/controllers/main.py

```diff
@@ -40,24 +40,27 @@ def view(self, db, token, action, id, view='calendar'):
             # Since we are in auth=none, create an env with SUPERUSER_ID
             env = Environment(cr, SUPERUSER_ID, {})
             attendee = env['calendar.attendee'].search([('access_token', '=', token)])
+            if attendee.event_id.id != int(id):
```

Comment:
Ensure that you're accessing the correct event. Previously, with one event invitation you gained access to all of Odoo's events as there was no verification the attendee token was indeed of the event you were looking for.